modules/metastorage/src/testFixtures/java/org/apache/ignite/internal/metastorage/server/SimpleInMemoryKeyValueStorage.java

```diff
@@ -359,21 +360,30 @@ public StatementResult invoke(If iif) {
         }
     }
 
-    /** {@inheritDoc} */
     @Override
     public Cursor<Entry> range(byte[] keyFrom, byte[] keyTo, boolean includeTombstones) {
         synchronized (mux) {
             return new RangeCursor(keyFrom, keyTo, rev, includeTombstones);
         }
     }
 
-    /** {@inheritDoc} */
     @Override
     public Cursor<Entry> range(byte[] keyFrom, byte[] keyTo, long revUpperBound, boolean includeTombstones) {
         return new RangeCursor(keyFrom, keyTo, revUpperBound, includeTombstones);
     }
 
-    /** {@inheritDoc} */
+    @Override
+    public Cursor<Entry> prefix(byte[] prefix, boolean includeTombstones) {
+        synchronized (mux) {
```

Comment:
Maybe you should add `synchronized` for `prefix`* methods header, for example `public synchronized Cursor<Entry> prefix(...)`?

Reply:
How would that work? All other methods use a separate object for synchronisation

Reply:
Write a comment why `mux` is used there.